code/obj/machinery/door/airlock.dm

```diff
@@ -2155,4 +2176,50 @@ TYPEINFO(/obj/machinery/door/airlock)
 						src.attach_signaler(which_wire+1, usr)
 						. = TRUE
 
+/obj/machinery/door/airlock/proc/user_demagging(mob/user, obj/item/current_tool)
+	. = FALSE
+
+	if(!user.traitHolder.hasTrait("training_engineer")) // If user doesn't have Engineer Training trait, simply return FALSE
+		return
+
+	switch(src.demag_status)
+		if(UNSCREW_STAGE)
```

Comment:
interferes with the ability to open the panel of emag doors normally 

prevents players from taking actions like fixing the power or de-electrifying

Reply:
I'm not really sure if this is a problem, specially when this action cannot be re-done. Once unscrewed, the de-emag stage going to CUT_WIRE_STAGE, which means it can be normally screwed and un-screwed after that, by anyone.